### DISCUSSION TITLE
Can Lustre be used as a backend file storage for the versity gateway?

### DISCUSSION THREAD

**Discussion by doublewood123** [versity/versitygw#815] [Q&A] [answered]
We have deployed a set of Lustre storage, and now the user requests an object interface based on this storage, and requests that files written through the Versity gateway can also be accessed by the Lustre client. Can it be done?

**Comment by benmcclelland** [comment, marked as answer]
@doublewood123 Yes, I believe Lustre should work fine.  When deploying the gateway with posix backend, the only requirement on the filesystem is that it has xattr support. The files can be seen through Lustre client as well. See this for the directory structure when accessing through object: https://github.com/versity/versitygw/wiki/POSIX-Backend#object-name-mapping. The top level directory exported through the gateway will treat all subdirectories as buckets.

Assuming the gateway works ok on Lustre, there is another advantage that you can load balance across multiple gateways on multiple hosts if you want as well. Each gateway is completely stateless. So if you setup multiple gateways hosting the same top level directory, you can use any gateway to handle incoming requests. To distribute load across the gateways, you can use a load balancer such as Pound proxy or HA proxy on the clients, or some hardware load balancer in front of the gateways. Or you can just use DNS to help spread out requests across systems.

**Reply by doublewood123** [reply]
thanks
